core/src/nozzle_subgraph/runner/reorg_handler.rs

```diff
@@ -0,0 +1,162 @@
+use alloy::primitives::{BlockHash, BlockNumber};
+use anyhow::anyhow;
+use futures::{future::try_join_all, StreamExt, TryFutureExt};
+use graph::{
+    blockchain::block_stream::FirehoseCursor,
+    nozzle::{
+        client::{LatestBlockBeforeReorg, RequestMetadata, ResponseBatch, ResumeStreamingQuery},
+        common::Ident,
+        Client,
+    },
+};
+use itertools::Itertools;
+use slog::debug;
+
+use super::{Compat, Context, Error, LatestBlocks};
+
+pub(super) async fn check_and_handle_reorg<NC>(
```

Comment:
This is really important. It would be good to have a comment on how this works and why. It might also be worth explaining what the other two methods here do.